packages/gatsby-plugin-google-gtag/src/gatsby-browser.js

```diff
@@ -29,3 +31,13 @@ exports.onRouteUpdate = ({ location }) => {
 
   return null
 }
+
+exports.onRenderBody = ({ setHeadComponents }) => {
+  setHeadComponents([
+    <link
+      rel="dns-prefetch"
+      key="dns-prefetch-google-analytics"
+      href="https://www.google-analytics.com"
+    />,
+  ])
+}
```

Comment:
Let's also add a preconnect so the tls handshake is done, dns prefetch is less important but is a necessary fallback for old browsers.

```suggestion
exports.onRenderBody = ({ setHeadComponents }) => {
  setHeadComponents([
    <link
      rel="preconnect"
      key="gatsby-plugin-gtag-preconnect"
      href="https://www.google-analytics.com"
    />,
    <link
      rel="dns-prefetch"
      key="gatsby-plugin-gtag-prefetch"
      href="https://www.google-analytics.com"
    />,
  ])
}
```